### ISSUE TITLE
[v8.0.4] Initial Values not available in Field componentWillMount

### ISSUE THREAD

**Issue by lewisheadden** [erikras/redux-form#4304]
### Are you submitting a **bug report** or a **feature request**?
Bug Report

### What is the current behavior?
In `componentWillMount` for `Field` components, field values are not set to their initialValues. See that in the `console.log` from `componentWillMount` in the sandbox the field value is `""`.

### What is the expected behavior?
In 7.4.2, initial values were available in `Field` components during the `componentWillMount` lifecycle. See that the `console.log` now reads `jim` as per the provided value.

### Sandbox Link
Redux Form 8.0.4: https://codesandbox.io/s/0yvykqqxwl
Redux Form 7.4.2: https://codesandbox.io/s/p423x6y30

### What's your environment?
Redux Form 8.0.4

### Other information
Please let me know if you need any more clarity and thanks for all the work you do!

**Comment by nerfologist** [comment]
I believe this is also the cause of my issue:

I am testing the upgrade to redux-form 8.0.4 using jest and enzyme's `mount` to mount a connected redux form with `initialValues`.

I am getting PropTypes warnings that required values are the wrong type (required boolean fields being passed empty strings). Logging renders I see a first render with empty string values, followed by a second render with the correct values. The first render seems to be generating the warning.

**Comment by tomasztomys** [comment]
the same error with sync validation and array... I don't get initial value of array to validate function in first render

**Comment by DavyJohnes** [comment]
@erikras Hi, any thoughts on this? May be any temp workaround? For my use cases redux-form v8 is completely broken because of this issue.

**Comment by jjosef** [comment]
componentWillMount is deprecated

**Comment by lewisheadden** [comment]
@jjosef You are correct. However, to clarify, moving this call to a constructor (as per [the deprecation notification](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html)) shows the same problem. 

If there's something else you think I should be doing, please let me know!

**Comment by srisonti** [comment]
I'm running into this same issue - I'm seeing the initialize action dispatched to Redux, the store is showing the correct values for both the `values` and `initial` keys, but then the individual Field components are all getting empty strings as values (however, if I inspect their `meta` props, the `meta.initial` prop is showing the correct initial value that the field _should_ be using).

Has anyone had any luck getting around this?

**Comment by MartijnHols** [comment]
`initialValues` don't seem to be loaded into the redux form state until after an initial render. Several of my components rely on there being an initial value so I made a helper decorator for a quick fix for those components so I don't have to rewrite all my components, just decorate them. The decorator simply doesn't render the form until it's initialized.

`reduxFormRequireInitialized.js`
```jsx
import React from 'react';

export default Component => (
  props => {
    if (!props.initialized) {
      return null;
    }

    return (
      <Component {...props} />
    );
  }
);

```

**Comment by alxmiron** [comment]
> The decorator simply doesn't render the form until it's initialized

I suppose this can lead to blinks or UI shifts

**Comment by kgram** [comment]
I've just encountered this as well. I think the problem is that react-redux has slightly changed behaviour after switching to the new context API. The initialize action is fired and the store is updated before rendering the connected field, but react-redux serves up a snapshot from when the render began. While this is annoying in this case, it does seem like a more correct behaviour.

I think it can be very simply fixed by reading `value` from `initialValues` instead of `formState` before initialization, a one-line change. I'll put together a pull-request.

**Comment by GuillotJessica** [comment]
first sandBox will work better with constructor like this :
```class TestComponent extends React.Component {
  constructor(props) {
    super(props)
    console.log(this.props.input.value);
  }

  render() {
    return null;
  }
}```
then modal doesn't render an empty string

**Comment by lynxtaa** [comment]
Issue is gone after updating to react-redux@7.0.0-beta.0

**Comment by pwhipp** [comment]
Upgrading to redux-form@8.1.0 and react-redux@6.0.1 appears to have introduced this problem into my application.
Upgrading to redux-form@8.2.1 and react-redux@7.0.3 does not appear to have got rid of it :(

**Comment by eugensunic** [comment]
Any news on that facing the same issues...

**Comment by Levino** [comment]
@eugensunic Please read the deprecation [notice](https://github.com/redux-form/redux-form#%EF%B8%8F-attention-%EF%B8%8F) and consider using final form...

**Comment by eugensunic** [comment]
@Levino I already have , currently on 7.4.2 and the next step in migration only.

**Comment by darekkay** [comment]
This _might_ be resolved with [8.3.10](https://github.com/redux-form/redux-form/releases/tag/v8.3.10). From a first look, I didn't encounter the problem anymore.
